<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1240</width>
    <height>840</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1240</width>
    <height>840</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1240</width>
    <height>840</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="s21::GLWidget" name="OGLWidget" native="true">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>20</y>
      <width>601</width>
      <height>601</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_open">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>81</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Open</string>
    </property>
    <property name="default">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_translation">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>281</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Translation</string>
    </property>
    <widget class="QScrollBar" name="scrollBar_translation_x">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-10000</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QScrollBar" name="scrollBar_translation_y">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>80</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-10000</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QScrollBar" name="scrollBar_translation_z">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>120</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-10000</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_translation_x">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_translation_y">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_translation_z">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>120</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_translation_x">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>40</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="minimum">
      <number>-10000</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_translation_y">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>80</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="minimum">
      <number>-10000</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="stepType">
      <enum>QAbstractSpinBox::DefaultStepType</enum>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_translation_z">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>120</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="minimum">
      <number>-10000</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_rotate">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>281</width>
      <height>421</height>
     </rect>
    </property>
    <property name="title">
     <string>Rotate</string>
    </property>
    <widget class="QLabel" name="label_rotate_x">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_y">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>210</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_z">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>330</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QDial" name="dial_y">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>160</y>
       <width>131</width>
       <height>121</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>359</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="pageStep">
      <number>10</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="tracking">
      <bool>true</bool>
     </property>
     <property name="invertedAppearance">
      <bool>true</bool>
     </property>
     <property name="wrapping">
      <bool>true</bool>
     </property>
     <property name="notchTarget">
      <double>1.000000000000000</double>
     </property>
     <property name="notchesVisible">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QDial" name="dial_z">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>280</y>
       <width>131</width>
       <height>121</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>359</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="pageStep">
      <number>10</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="tracking">
      <bool>true</bool>
     </property>
     <property name="invertedAppearance">
      <bool>true</bool>
     </property>
     <property name="invertedControls">
      <bool>false</bool>
     </property>
     <property name="wrapping">
      <bool>true</bool>
     </property>
     <property name="notchTarget">
      <double>1.000000000000000</double>
     </property>
     <property name="notchesVisible">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QDial" name="dial_x">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>40</y>
       <width>131</width>
       <height>121</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>359</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="pageStep">
      <number>10</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="tracking">
      <bool>true</bool>
     </property>
     <property name="invertedAppearance">
      <bool>true</bool>
     </property>
     <property name="wrapping">
      <bool>true</bool>
     </property>
     <property name="notchTarget">
      <double>1.000000000000000</double>
     </property>
     <property name="notchesVisible">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_rotate_x">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>90</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="wrapping">
      <bool>true</bool>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>359</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_rotate_y">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>210</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="wrapping">
      <bool>true</bool>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>359</number>
     </property>
     <property name="stepType">
      <enum>QAbstractSpinBox::DefaultStepType</enum>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_rotate_z">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>330</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="wrapping">
      <bool>true</bool>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>359</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_typeofprojection">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>40</y>
      <width>281</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>Type of projection</string>
    </property>
    <widget class="QRadioButton" name="radioButton_orthogonal">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>30</y>
       <width>99</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>orthogonal</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioButton_perspective">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>99</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>perspective</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_information">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>620</y>
      <width>601</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Information</string>
    </property>
    <widget class="QLabel" name="label_name_info">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>581</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Name: empty</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_path_info">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>581</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Full path: empty</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_vertices_info">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>581</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Vertices: 0</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_edges_info">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>581</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Edges: 0</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_setting_verciets">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>110</y>
      <width>281</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Setting vertices</string>
    </property>
    <widget class="QComboBox" name="comboBox_vertices">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>30</y>
       <width>281</width>
       <height>32</height>
      </rect>
     </property>
     <property name="editable">
      <bool>false</bool>
     </property>
     <property name="currentText">
      <string>disable</string>
     </property>
     <property name="currentIndex">
      <number>2</number>
     </property>
     <item>
      <property name="text">
       <string>circle</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>square</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>disable</string>
      </property>
     </item>
    </widget>
    <widget class="QScrollBar" name="scrollBar_vertices">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>70</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>1</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
     <property name="sliderPosition">
      <number>1</number>
     </property>
     <property name="tracking">
      <bool>false</bool>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_vertices_size">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>size</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_vertices">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>70</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="wrapping">
      <bool>false</bool>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>0</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_setting_edges">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>220</y>
      <width>281</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Setting Edges</string>
    </property>
    <widget class="QComboBox" name="comboBox_edges">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>30</y>
       <width>281</width>
       <height>32</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>solid</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>dashed</string>
      </property>
     </item>
    </widget>
    <widget class="QScrollBar" name="scrollBar_edges">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>70</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_edges_size">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>size</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_edges">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>70</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="wrapping">
      <bool>false</bool>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_setting_color">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>330</y>
      <width>281</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Setting Color</string>
    </property>
    <widget class="QPushButton" name="pushButton_bg">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>201</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Background</string>
     </property>
    </widget>
    <widget class="QFrame" name="frame_bg">
     <property name="geometry">
      <rect>
       <x>219</x>
       <y>30</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame {
	background-color: black;
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_vertices">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>201</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Vertices</string>
     </property>
    </widget>
    <widget class="QFrame" name="frame_vertices">
     <property name="geometry">
      <rect>
       <x>219</x>
       <y>70</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame {
	background-color: white;
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
    <widget class="QFrame" name="frame_edges">
     <property name="geometry">
      <rect>
       <x>219</x>
       <y>110</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame {
	background-color: white;
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_edges">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>201</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Edges</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_scale">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>620</y>
      <width>281</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Scale</string>
    </property>
    <widget class="QScrollBar" name="rollBar_scale">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>10</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="singleStep">
      <number>10</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_scale">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>40</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="buttonSymbols">
      <enum>QAbstractSpinBox::NoButtons</enum>
     </property>
     <property name="minimum">
      <number>10</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="singleStep">
      <number>10</number>
     </property>
     <property name="stepType">
      <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
     </property>
     <property name="value">
      <number>100</number>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_default">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>710</y>
      <width>281</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Default settings</string>
    </property>
   </widget>
   <widget class="QComboBox" name="formatBox">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>510</y>
      <width>161</width>
      <height>32</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>.jpeg</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>.bmp</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="pushButton_save">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>510</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Save</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_record">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>550</y>
      <width>261</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Record</string>
    </property>
   </widget>
   <zorder>pushButton_open</zorder>
   <zorder>OGLWidget</zorder>
   <zorder>groupBox_translation</zorder>
   <zorder>groupBox_rotate</zorder>
   <zorder>groupBox_typeofprojection</zorder>
   <zorder>groupBox_information</zorder>
   <zorder>groupBox_setting_verciets</zorder>
   <zorder>groupBox_setting_edges</zorder>
   <zorder>groupBox_setting_color</zorder>
   <zorder>groupBox_scale</zorder>
   <zorder>pushButton_default</zorder>
   <zorder>formatBox</zorder>
   <zorder>pushButton_save</zorder>
   <zorder>pushButton_record</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1240</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHistory">
    <property name="title">
     <string>History</string>
    </property>
   </widget>
   <addaction name="menuHistory"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSave_gif">
   <property name="text">
    <string>Save gif</string>
   </property>
  </action>
  <action name="actionSave_screenshot">
   <property name="text">
    <string>Save image</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
